### ISSUE TITLE
Routes (middleware) is firing multiple times

### ISSUE THREAD

**Issue by talmog** [expressjs/generator#169]
Hi, I am using the express generator with hbs view.
When I modify the layout.hbs to include some assets the routes are firing multiple times:

I added console.log to the routes/index.js file:
```
var express = require('express');
var router = express.Router();

var counter = 0;
/* GET home page. */
router.get('/', function(req, res, next) {
  console.log('COUNTER: ' + counter);
  counter++;
  res.render('index', { title: 'Express' });
});

module.exports = router;
```

and in my console I am getting:
```
COUNTER: 0
GET / 200 106.119 ms - 3288
COUNTER: 1
GET / 304 6.183 ms - -
COUNTER: 2
GET / 304 8.905 ms - -
GET /stylesheets/style.css 200 43.838 ms - -
GET /images/phonedirectory-l.png 304 3.003 ms - -
COUNTER: 3
GET / 304 13.946 ms - -
COUNTER: 4
GET / 304 10.130 ms - -
GET /images/favicon-32x32.png 304 0.938 ms - -
COUNTER: 5
GET /images/favicon-16x16.png 304 0.531 ms - -
GET / 304 6.982 ms - -
GET /images/favicon-96x96.png 304 0.382 ms - -
GET /images/android-icon-192x192.png 304 0.642 ms - -
GET /images/favicon.ico 304 0.284 ms - -
GET /images/favicon.ico 304 0.444 ms - -
COUNTER: 6
GET / 304 7.536 ms - -
GET /stylesheets/style.css.map 200 3.727 ms - 25069
GET /serviceworker.js 404 6.650 ms - 4354
```

This is a big issue since I want to make an API call in one of my routes,and don't want to make it 6 times.
This number depends on the view. If I remove the code from layout.hbs it only fires once.

here is my layout code
```
<!doctype html>
<html amp lang="en">
<head>
  <meta charset="utf-8">
  <script async src="https://cdn.ampproject.org/v0.js"></script>
  <title>{{title}}</title>
  <link rel="canonical" href="/">
  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
  <link href="https://fonts.googleapis.com/css?family=Source+Sans+Pro:300,400,700" rel="stylesheet">
  <link rel='stylesheet' href='stylesheets/style.css'/>
  {{#if homepage}}
    <link rel='stylesheet' href='/stylesheets/home.css'/>
  {{/if}}

  <link rel="apple-touch-icon" sizes="57x57" href="/images/apple-icon-57x57.png">
  <link rel="apple-touch-icon" sizes="60x60" href="/images/apple-icon-60x60.png">
  <link rel="apple-touch-icon" sizes="72x72" href="/images/apple-icon-72x72.png">
  <link rel="apple-touch-icon" sizes="76x76" href="/images/apple-icon-76x76.png">
  <link rel="apple-touch-icon" sizes="114x114" href="/images/apple-icon-114x114.png">
  <link rel="apple-touch-icon" sizes="120x120" href="/images/apple-icon-120x120.png">
  <link rel="apple-touch-icon" sizes="144x144" href="/images/apple-icon-144x144.png">
  <link rel="apple-touch-icon" sizes="152x152" href="/images/apple-icon-152x152.png">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-icon-180x180.png">
  <link rel="icon" type="image/png" sizes="192x192" href="/images/android-icon-192x192.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="96x96" href="/images/favicon-96x96.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16.png">
  <link rel="shortcut icon" href="/images/favicon.ico" type="image/x-icon">
  <link rel="icon" href="/images/favicon.ico" type="image/x-icon">
  <link rel="manifest" href="/manifest.json">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="msapplication-TileImage" content="/images/ms-icon-144x144.png">
  <meta name="theme-color" content="#ffffff">

  <script async custom-element="amp-form" src="https://cdn.ampproject.org/v0/amp-form-0.1.js"></script>

</head>
<body>
<header class="row">
  <div class="small-6 columns"><a href="/">
    <amp-img src="/images/logo.png" alt="Phone Directory" height="50" width="123"></amp-img>
  </a></div>
  <div class="small-6 columns">
    <div class=" text-right aligner-bottom">
      <a href="/about" class="menu-links">About</a>
      <a href="/contact" class="menu-links">Contact</a>
    </div>
  </div>
</header>
{{{body}}}

</body>
</html>

```

Thank you!

**Comment by dougwilson** [comment]
Hi @talmog that's really weird, and I haven't ever heard of that before. I tried out your code on top of the generated Express and the `COUNTER` just did 1 and never again. All I did was load `http://localhost:3000/` in Google Chrome. Is that the same thing you did? Is it possible to set a network capture of all the requests to see what's happening? Perhaps even change the `morgan` to use `apache` instead of `dev` to get the IP addresses listed in the log just in case something else is hitting your server at the same time?

**Comment by talmog** [comment]
I will try, I tried again from a fresh copy of the generator, and I still get the same result. did you include the layout and some static files?

**Comment by dougwilson** [comment]
Yea, I included the layout you provided above, and it was rendered, though a bunch of the things it referenced I didn't have, so I just made up my own `logo.png`, my own `manifest.json`, my own `home.css`, etc. If you think that those may be contributing, I would need a complete copy of everything to try it out, but I tried with as much info that you provided so far without being able to replicate your results, so I'm not sure how to proceed. Let me know if you have any thoughts for what I either (a) need to replicate the issue or (b) a diagnosis of what is wrong in the generator that is causing the issue.
